volumeData.py

```diff
@@ -0,0 +1,357 @@
+            Posegmentowana macierz 3D: VolumeData
+        """
+        thresh = 0
+        threshList = []
+        for i in range(iterCount):
+            thresh = lib.threshOtsu(self._data3D, thresh)
+            threshList.append(thresh)
+        
+        segVolume = self.segmentDataByThresholds(threshList)
+        print('Znalezione prgi: ', threshList)
+        
+        return segVolume
+
+    def regionGrowingSegment(self, c=2, sigma0=20):
+        """
+        Wykonuje segmentację algorytmem Region Growing. Za punkt startowy bierze punkt z obszaru zsegmentowantego progowaniem Otsu.
+        Wartość zwracana
+            Posegmentowana macierz 3D: VolumeData
+        """
+        img = self._data3D
+
+        trzyDe = True
+        if len(img.shape) == 2:
+            trzyDe = False
+
+        thresh = threshold_yen(img)
+        bw = img >= thresh
+        label_image = label(bw)
+
+        centroidy = []
+        for region in regionprops(label_image):
+            centroidy.append(region.centroid)
+
+        centroidy = np.array(centroidy)
+        startPoints = np.around(centroidy).astype(int)
+        regions = []
+        for startPoint in startPoints:
+
+            if startPoint is None:
+                if trzyDe:
+                    startPoint = [0,0,0]
+                else:
+                    startPoint = [0,0]
+
+            # Parametr algorytmu
+            c = 2
+
+            region = [startPoint]
+
+            # Transformacje, które pozwolą na multiindeksowanie macierzy/obrazu
+            regionArr = np.array(region)
+            if trzyDe:
+                tempImg = img[regionArr[:,0], regionArr[:,1], regionArr[:,2]]
+            else:
+                tempImg = img[regionArr[:,0], regionArr[:,1]]
+
+            avg = np.mean(tempImg)
+            sigma0 = 20
+            sigmaC = sigma0
+
+            licznikTrafien = 0
+            R = 1
+
+            # Pętla po pikselach/wokselach. Wychodząc z punktu startPoint promieniście rozchodzi się algorytm.
+            while(True):
+                print(f'Promień: {R}')
+                points = lib.pointsInRadius(startPoint, R)
+                if not lib.belongsToArr(points, img):    # Sprawdzenie, czy przynajmniej jeden punkt należy do macierzy
+                    print('Wyjście poza macierz wszystkich punktów.')
+                    break
+
+                licznikTrafien = 0
+                for p in points:
+                    # p - współrzędne punktu 
+                    if lib.belongsToArr(p, img):
+                        print(f'Punkt {p}: {img[p[0], p[1], p[2]]}')
+                        if trzyDe:
+                            tempWynik = abs(img[p[0], p[1], p[2]] - avg)
+                        else:
+                            tempWynik = abs(img[p[0], p[1]] - avg)
+                        print(f'tempWynik: {tempWynik}, sigmaC: {sigmaC}')
+
+                        if tempWynik <= sigmaC:
+                            region.append(p)
```

Comment:
Wadą tego algorytmu jest to że potrafi generować obszary które są nieciągłe. Dla przykładu:
```
000010000
000010000
000010000
000010000
000000000
000000000
000000000
000010000
```

Startując z czwartego rzędu i 5 kolumny algorytm złapie wszystkie jedynki.

O ile czyni go to porównywalnym z progowaniem Otsu czy innym to tracimy zaletę region growing która powinna ograniczyć przestrzeń wyszukiwania do obszaru ciągłego.

Przy bardzo dużych obrazach i małych regionach można by pomyśleć o macierzach rzadkich. Ale z tego co widzę może to wymagać więcej pracy niż to warto.

Myślę że wystarczy pracować na masce binarnej:
- zaczynamy od maski gdzie wszystkie elementy są `False` poza punktem startowym
- na niej wykonujemy operację `skimage.morphology.binary_dilation`
- można porównać z poprzednim obrazem żeby dostać maskę nowych punktów
- dla nowych punktów które wyciągamy ich indeksy i sprawdzamy czy ich intensywność nie różni się średniej o więcej niż sigma. Usuwamy z maski te które mają większą różnicę
- żeby efektywnie liczyć średnią można wykorzystać twoją implementację z zbiorem wartości intensywności punktów. W pierwszym kroku wyciągamy indeksy z maski binarnej dla wartości True w drugim kroku indeksujemy obraz i wyciągamy wartości intensywności.